### ISSUE TITLE
Why does SubOrchestrator take a few seconds to start?

### ISSUE THREAD

**Issue by hiroyuki-1001** [Azure/azure-functions-durable-extension#1110]
Compiled C#
Microsoft.Azure.WebJobs.Extensions.DurableTask: 1.8.4
AzureFunctionsVersion: v2
Function App: 
--Scale up: S1
--Scale out: Manual scale 1 or 2

In case of 1 instance, the SubOrchestrator will start within 1 second.
In case of 2 instances, it may take a few seconds to start the SubOrchestrator.

Why does it take a few seconds to start a SubOrchestrator with 2 instances?

logs:
(1 instance)
![image](https://user-images.githubusercontent.com/58962204/70966871-f04f2e80-20d7-11ea-8cd7-a58ccc7a59fa.png)

(2 instances)
![image](https://user-images.githubusercontent.com/58962204/70966895-fba25a00-20d7-11ea-82f7-8cbbaaf09aae.png)

sample-code:
https://github.com/hiroyuki-1001/sample/blob/master/FunctionApp7/Function1.cs

**Comment by hiroyuki-1001** [comment]
2 instances log (Without filter)
![image](https://user-images.githubusercontent.com/58962204/70967461-a6ffde80-20d9-11ea-87fa-cccde422fe48.png)

**Comment by cgillum** [comment]
Hi @hiroyuki-1001. This is expected behavior. The delay is because of exponential-backoff queue polling. When you have one app instance, there is an optimization where we can signal the orchestration to start running immediately without waiting for polling. However, if you have multiple app instances, sometimes the orchestration will start on a different VM and we must wait for the second instance to poll for the start message.

If all app instances are busy, then the queue polling delay goes away since we use exponential backoff and the backoff will be very small or zero.

**Comment by hiroyuki-1001** [comment]
@cgillum 
Is there a way to reduce the queue polling delay for multiple instances?

In the case of 2 instances, we improved a little by reducing the control queue,
If the number of instances was 3, this method could not be improved.

**Comment by olitomlinson** [comment]
@hiroyuki-1001 

The answer to your question lies here

https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-perf-and-scale#queue-polling

**Comment by hiroyuki-1001** [comment]
@olitomlinson 

I've tried the settings in the past and found that the settings slow down as the number of instances increases.

Is there anything else I can do?

**Comment by olitomlinson** [comment]
@hiroyuki-1001 Sorry I'm struggling to understand clearly. Please can you clarify what you mean when you say the settings slow down as the number of instances increase?

If you can provide examples with statistics, that would be helpful!

**Comment by hiroyuki-1001** [comment]
@olitomlinson  Sorry. Explanation was wrong.
I checked with 2 instances.
I changed the polling setting to 10 seconds, but the launch of the SubOrchestrator was not fast.

**Comment by cgillum** [comment]
@hiroyuki-1001 When you changed the polling setting to 10 seconds, did the sub-orchestrator launch in 10 seconds or less? This is the expectation. `maxQueuePollingInterval` is basically the _maximum delay_ in starting an orchestrator or activity function.

**Comment by hiroyuki-1001** [comment]
@cgillum The sub-orchestrator did not start within 10 seconds.Does that mean there is no workaround for this?

**Comment by cgillum** [comment]
@hiroyuki-1001 what is the instance ID of your orchestration?

**Comment by hiroyuki-1001** [comment]
@cgillum instance ID is d4f2d54f991d43a7b612787be3fdeb90:2.

**Comment by cgillum** [comment]
If changing the polling interval didn't help, then there is likely some kind of performance issue. One explanation is that there are too many orchestrations running and queue processing is taking a long time. Is that a possibility for your case?

The other possibility is that there are other internal problems causing queue processing to slow down. Some other users did notice similar problems recently, and we've published a fix. Can you try upgrading to [Microsoft.Azure.WebJobs.Extensions.DurableTask v1.8.5](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.5) and let us know if you still observe this problem?

**Comment by hiroyuki-1001** [comment]
Thank you very much.

> Is it possible in your case?
In my case, I call the sub-orchestrator multiple times. (The Top Orchestrator calls it only once.)

I will check it with version 1.8.5.

**Comment by itsyashpal** [comment]
> If changing the polling interval didn't help, then there is likely some kind of performance issue. One explanation is that there are too many orchestrations running and queue processing is taking a long time. Is that a possibility for your case?
> 
> The other possibility is that there are other internal problems causing queue processing to slow down. Some other users did notice similar problems recently, and we've published a fix. Can you try upgrading to [Microsoft.Azure.WebJobs.Extensions.DurableTask v1.8.5](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.5) and let us know if you still observe this problem?

@cgillum how can we mitigate issue if we have too many orchestrations running? We have a durable function which is expected to called 1 Million times a day and we are seeing the delay in starting the function after trigger. We do not think it is due to long poll as we are getting continuous requests through out the day ..

**Comment by mpaul31** [comment]
> > If changing the polling interval didn't help, then there is likely some kind of performance issue. One explanation is that there are too many orchestrations running and queue processing is taking a long time. Is that a possibility for your case?
> > The other possibility is that there are other internal problems causing queue processing to slow down. Some other users did notice similar problems recently, and we've published a fix. Can you try upgrading to [Microsoft.Azure.WebJobs.Extensions.DurableTask v1.8.5](https://github.com/Azure/azure-functions-durable-extension/releases/tag/v1.8.5) and let us know if you still observe this problem?
> 
> @cgillum how can we mitigate issue if we have too many orchestrations running? We have a durable function which is expected to called 1 Million times a day and we are seeing the delay in starting the function after trigger. We do not think it is due to long poll as we are getting continuous requests through out the day ..

did you try increasing the max concurrency setting for orchestrations?

**Comment by dsm0880** [comment]
@mpaul31 and @itsyashpal I'm having a similar issue. @cgillum We run 10 of thousands of orchestrations and activity triggers a day, which I feel like should be handled with ease. What is interesting is by simply "changing" the max concurrency of activity triggers or orchestrations it seems to cycle the durable functions and it starts picking things up very quickly, but within about 30-40 seconds everything haults again. I have a max concurrency on both activity triggers and orchestrations set to 1000 (much higher than we need) with orchestrations still hanging for processing.

**Comment by cgillum** [comment]
@dsm0880 10K orchestrations/activities per day should not be a problem. This issue tracks delays in starting new instances, but if you're observing that instances are completely stopped, then that's a different problem. Can you open a new issue and provide us with an instance ID, a timeframe, and the region you're running in?

**Comment by dsm0880** [comment]
@cgillum Will do

**Comment by hiroyuki-1001** [comment]
@cgillum 

in v1.8.5
We have confirmed that the sub-orchestrator starts within 10 seconds.

Thank you very much.

**Comment by hiroyuki-1001** [comment]
Are ContinueAsNew issues remaining in v1.8.5? (#1374)

**Comment by cgillum** [comment]
@hiroyuki-1001 yes, the `ContinueAsNew` issue was fixed in v1.8.5. The fix is in the dependent [DurableTask.AzureStorage v1.7.2](https://github.com/Azure/durabletask/releases/tag/durabletask.azurestorage-v1.7.2) package.

**Comment by hiroyuki-1001** [comment]
@cgillum Thank you very much.

**Comment by cgillum** [comment]
For anyone following this issue, I've published updated documentation that describes orchestration start delays. You can find it here: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-perf-and-scale#orchestration-start-delays (and eventually it will be translated into other languages, like Japanese, soon)
